### ISSUE TITLE
2.5.0 GraphQL error when invoking action with a relationship: "a join column is missing from the response"

### ISSUE THREAD

**Issue by Jrodseth** [hasura/graphql-engine#8399]
### Version Information

Server Version: 2.5.0
CLI Version (for CLI related issue):

### Environment
OSS

### What is the expected behaviour?
Calling an action with a relationship should return without an error.

### Keywords
2.5.0, a join column is missing from the response, unexpected

### What is the current behaviour?
When an action with a relationship is called, a graphql error is returned by the server. Apollo client resolves the error to:
```
{
  "graphQLErrors": [
    {
      "extensions": {
        "path": "$",
        "code": "unexpected"
      },
      "message": "a join column is missing from the response: <related_table_id>_join_column"
    }
  ],
  "clientErrors": [],
  "networkError": null,
  "message": "a join column is missing from the response: <related_table_id>_join_column",
}

```
where <related_table_id> is the name of the field which joins the relationship.

### How to reproduce the issue?

1. Create an action in < 2.5.0 with a relationship
2. Update to 2.5.0
3. Invoke the action

### Any possible solutions?

Potential regression / bug introduced in 2.5.0

### Can you identify the location in the source code where the problem exists?

Still investigating

### If the bug is confirmed, would you be willing to submit a PR?

Maybe

**Comment by tirumaraiselvan** [comment]
Hi @Jrodseth 

Does the issue go away if you go back to v2.4.0? Would you be able to help us with a repro? You can either share your metadata with the action definition, types and postgres table schema or construct a minimal example.

I tried the following and not facing the reported error:

This is my action:

```
type Mutation {
  actionName(
    arg1: SampleInput!
  ): SampleOutput
}

input SampleInput {
  username: String!
  password: String!
}

type SampleOutput {
  accessToken: String!
  name: String!
}

```

I created a relationship from table `user (id, name, email)` to the `name` field in `SampleOutput`:

![image](https://user-images.githubusercontent.com/8663570/163362223-d76f7cc5-4309-47f8-9bef-6d325b738848.png)


I can run a query as follows without any errors:

```
mutation {
  actionName(arg1: {password: "adfa", username: "adfa"}) {
    accessToken
    actionRelationship {
      email
    }
  }
}
```

![image](https://user-images.githubusercontent.com/8663570/163362636-7e012007-c4db-4969-a587-1619c6c49d11.png)

**Comment by Jrodseth** [comment]
Hey @tirumaraiselvan thanks for looking into this.

Yes, we saw the issue go away when we reverted to 2.4.0. I'll scrub some of our metadata and put it into a minimal example today.

**Comment by isilweo** [comment]
Hey, 
just wanted to confirm this error. Reverting back to 2.4.0 helped

**Comment by tirumaraiselvan** [comment]
@Jrodseth Any luck with the repro? Happy to just get your metadata and take it from there. You can mail me at `tiru@hasura.io`

**Comment by LucaDe** [comment]
Same issue here on v2.5.0-cloud.2, works fine on v2.3.1-cloud.1

**Comment by Jrodseth** [comment]
@tirumaraiselvan thanks, I've been tight on time. I'll email you!

**Comment by 0x777** [comment]
@isilweo / @LucaDe Can you help us with a repro?

**Comment by happyfloat** [comment]
I was on vacation and just noticed that my main feature of my production app did broke because of the automatic cloud updates once again... PLEASE DONT AUTO UPDATE THE CLOUD VERSIONS!!!!

Is there a fix for the cloud version available? It's still not working on 2.5.1-cloud.1

Otherwise, thanks for the great tool and support 👍

**Comment by happyfloat** [comment]
I found the issue. It's a breaking change somewhat after 2.3.x, which should be documented @tirumaraiselvan.

I got an action result type defintion like this:
```gql
type UpsertEnrolmentOutput {
  id: uuid
  missing_requirements: jsonb
}
```

Both `id` and `missing_requirements` are optional and my action handler did only return one or the other field, which was working fine before. The `id` field has a related relationship. Now I'm returning `null` instead and it is working again.
